### PR TITLE
[ESNOOK-11] Add park functionality and endpoint authentication

### PR DESCRIPTION
**Pull request by WildBunnie**
In this pull request we:

- Added requirements.txt
- Added docker file and docker compose
- Refactored the client endpoint to use the cognito token
- Added the functionality to create a park
- Refactored the park endpoints to use the cognito token
- Added repository for the parks
- Changed the database models to better fit the current application (see [here](https://dbdiagram.io/d/67386f2be9daa85acaa80ceb))

### CODE REVIEW

**Comment by WildBunnie** [comment]
I changed the model of the database quite a bit, i wanted to know your opinions on whether something is missing or whether something needs changing.

**Comment by zegameiro** [comment]
About the database model, shouldn't exist a table for the parking spots?

File: src/models/park.py [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
-    location = Column("location", String)
-    total_spots = Column("total_spots", Integer)
-    owner = Column(Integer, ForeignKey("client.id"))
-
+    id = Column(Integer, Identity(start=1, increment=1), primary_key=True)
+    name = Column(String, nullable=False)
+    picture = Column(LargeBinary, nullable=True)
+    latitude = Column(Float, nullable=True)
+    longitude = Column(Float, nullable=True)
+    rtsp_url = Column(String, nullable=True)
```
Comment (fungame2270) [inline src/models/park.py:14]: The spots where suppose to be here no?

**Review by fungame2270 — COMMENTED** [review]
let the sonnarcloud know that docker its not a problem because its a dev one

**Comment by zegameiro** [comment]
> let the sonnarcloud know that docker its not a problem because its a dev one

Done!

File: src/models/park.py (hunk 1 above)
Comment (zegameiro) [reply src/models/park.py:14]: Yeah I think so, it should exist there a list of parking spots, and in the database diagram that @WildBunnie made there isn't an entity for the parking spot.

**Comment by WildBunnie** [comment]
> About the database model, shouldn't exist a table for the parking spots?

> The spots where suppose to be here no?

That's a separate issue on jira, this is just for adding parks

**Comment by fungame2270** [comment]
> > About the database model, shouldn't exist a table for the parking spots?
> 
> > The spots where suppose to be here no?
> 
> That's a separate issue on jira, this is just for adding parks

makes sense

**Review by fungame2270 — DISMISSED** [review]
lgtm

**Review by zegameiro — DISMISSED** [review]
Everything looks good to me just missing fix the tests for the client endpoints and good to go

**Review by zegameiro — APPROVED** [review]
LGTM

**Review by fungame2270 — APPROVED** [review]
lgtm

**Pull request merged**